src/core/stream/period/get_adaptation_switch_strategy.ts

```diff
@@ -64,22 +64,66 @@ export default function getAdaptationSwitchStrategy(
     return { type: "continue", value: undefined };
   }
 
-  // remove from that intersection what we know to be the right Adaptation
-  const adaptationInBuffer = getBufferedRangesFromAdaptation(queuedSourceBuffer,
-                                                             period,
-                                                             adaptation);
-  const { currentTime } = clockTick;
+  queuedSourceBuffer.synchronizeInventory();
+  const inventory = queuedSourceBuffer.getInventory();
+
+  // Continue if we have no other Adaptation buffered in the current Period
+  if (!inventory.some(buf => buf.infos.period.id === period.id &&
+                             buf.infos.adaptation.id !== adaptation.id))
```

Comment:
So, yea, U did the fix related to the Multi Period.

I will wait that MR to be merged into next to rebase my base branch. I will expect some merge conflict :)